src/lib/components/Swap/Input.tsx

```diff
@@ -27,31 +25,41 @@ interface InputProps {
 }
 
 export default function Input({ disabled }: InputProps) {
-  const input = useAtomValue(inputAtom)
-  const setValue = useUpdateInputValue(inputAtom)
-  const setToken = useUpdateInputToken(inputAtom)
-  const balance = mockCurrencyAmount
+  const {
+    currencyBalances: { [Field.INPUT]: balance },
+    currencyAmounts: { [Field.INPUT]: inputCurrencyAmount },
+  } = useSwapInfo()
+  const inputUSDC = useUSDCValue(inputCurrencyAmount)
+
+  const [typedInputAmount, updateTypedInputAmount] = useInputAmount()
+  const [inputCurrency, updateInputCurrency] = useInputCurrency()
+
+  //TODO(ianlapham): mimic logic from app swap page
+  const mockApproved = true
 
   return (
-    <InputColumn gap={0.5} approved={input.approved !== false}>
+    <InputColumn gap={0.5} approved={mockApproved}>
       <Row>
         <ThemedText.Subhead2 color="secondary">
           <Trans>Trading</Trans>
         </ThemedText.Subhead2>
       </Row>
       <TokenInput
-        input={input}
+        currency={inputCurrency}
+        amount={(typedInputAmount !== undefined ? typedInputAmount : inputCurrencyAmount?.toSignificant(6)) ?? ''}
         disabled={disabled}
-        onMax={balance ? () => setValue(1234) : undefined}
-        onChangeInput={setValue}
-        onChangeToken={setToken}
+        onMax={balance ? () => updateTypedInputAmount(balance.toExact()) : undefined}
+        onChangeInput={(val) => updateTypedInputAmount(val ?? '')}
```

Comment:
`onChangeInput=updateTypedInputAmount` - the parameters should match

Reply:
We cant set undefined as the`amount` value so have to use empty string if no value